pkg/plan/recipe/install_plans.go

```diff
@@ -184,10 +185,37 @@ func BuildCRIPlan(ctx context.Context, criSpec *existinginfrav1.ContainerRuntime
 	return &p
 }
 
+// BinInstaller creates a function to install binaries based on package type and cluster flavors
+func BinInstaller(pkgType resource.PkgType, f *eksd.EKSD) (func(string, string) plan.Resource, error) {
+	if f != nil {
+		log.Debugf("Using flavor %+v", f)
+		return func(binName, version string) plan.Resource {
+			// TODO (Mark) logic for the architecture
+			binURL, _, err := f.KubeBinURL(binName)
+			if err != nil {
+				log.Fatalf("%v", err)
+				return nil
+			}
+			// TODO Use the sha256 value to verify the downlaod
+			return &resource.Run{
+				Script:     object.String(fmt.Sprintf("curl -o /bin/%s %s && chmod 755 /bin/%s || true", binName, binURL, binName)),
```

Comment:
Do we want "true"? If the curl or chmod fails, shouldn't we punt?

Reply:
I think so.  I was following another run resource which used that.

Reply:
Some run resources do that because failing isn't important. Here, if we fail, we're dead in the water, no? I don't think we should go on and pretend everything is fine...